### ISSUE TITLE
resolveJsonModule transpile does not strip trailing commas

### ISSUE THREAD

**Issue by cspotcode** [Microsoft/TypeScript#24530]
**TypeScript Version:**  2.9.1

**Search Terms:** 

JSON
resolveJsonModule
json trailing comma

**Code**

```ts
import * as ts from 'typescript';
console.log(ts.version);
const output = ts.transpileModule(
    `
        {"foo": 123,} // note trailing comma
    `,
    {
        fileName: 'foo.json',
        compilerOptions: {
            resolveJsonModule: true,
            removeComments: true
        },
        reportDiagnostics: true
    }
);
console.dir(output.diagnostics); // no errors
console.dir(output.outputText); // '{ "foo": 123, }\r\n' still has trailing comma
```

**Expected behavior:**

Emitted JSON does not have trailing comma: `{ "foo": 123 }`
...or compiler emits a diagnostic to tell me the trailing comma is not allowed in input json.

**Actual behavior:**

Emitted JSON has trailing comma: `{ "foo": 123, }` and no diagnostics are reported.

**Playground Link:**  

N/A because Playground assumes .ts file extension.

**Related Issues:**

**Comment by mhegazy** [comment]
@sheetalkamat can you take a look please.
